crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -931,30 +931,28 @@ impl HolochainP2pHandler for HolochainP2pActor {
     fn handle_publish(
         &mut self,
         dna_hash: DnaHash,
-        from_agent: AgentPubKey,
+        _from_agent: AgentPubKey,
         request_validation_receipt: bool,
         dht_hash: holo_hash::AnyDhtHash,
         ops: Vec<(holo_hash::DhtOpHash, holochain_types::dht_op::DhtOp)>,
         timeout_ms: Option<u64>,
     ) -> HolochainP2pHandlerResult<()> {
+        use kitsune_p2p_types::KitsuneTimeout;
+
         let space = dna_hash.into_kitsune();
-        let from_agent = from_agent.into_kitsune();
         let basis = dht_hash.to_kitsune();
+        let timeout = match timeout_ms {
+            Some(ms) => KitsuneTimeout::from_millis(ms),
+            None => KitsuneTimeout::from_millis(30_000),
+        };
```

Comment:
FYI

```suggestion
        let timeout = KitsuneTimeout::from_millis(timeout_ms.unwrap_or(30_000));
```

Also can 30k be a const?

Reply:
switched to implicit timeout from tuning_params